src/lib/EDSL.mli

```diff
@@ -288,7 +288,58 @@ val tmp_file: ?tmp_dir: string t -> string -> file
 
 (** {3 Command Line Parsing} *)
 
+(** Typed command-line parsing for your shell scripts, à la {!Prtinf.scanf}. *)
 module Command_line: sig
+  (** 
+
+     Use this module like OCaml's {!Printf.scanf} function.
+
+     - Build a command-line “format specification” using the {!Arg} module.
+     - Call the {!parse} function with an appropriately typed function.
+
+     Example: {[
+       let cli_spec =
+         Command_line.Arg.(
+           string
+             ~doc:"The URL to the stuff" ["-u"; "--url"]
+             ~default:no_value
+           & flag ["-c"; "--all-in-tmp"] ~doc:"Do everything in the temp-dir"
```

Comment:
I would get rid of this flag, it is confusing. Where would you operate if not in the temp dir, the current dir? Seems like this variable would be better suited as a string. Perhaps a better argument for a flag (boolean option) would be something like remove intermediate target.